### PR TITLE
Computers

### PR DESCRIPTION
**Pull request by THATBLADEBOY**
# Description

Added the following functionality:

**Given** a user wants to view all computers
**When** the user clicks on the Computers item in the navigation bar
**Then** the user should see a list of all computers
**And** each item should be a hyperlink that can be clicked to view the details

**Given**  a user is viewing all computers
**When** the user clicks the Create New link
**Then** the user should be presented with a form in which the following information can be entered

1. Computer manufacturer
2. Computer make
3. Purchase date

**Given**  user is viewing a single computer
**When** the user clicks on the Delete link
**Then** the user should be presented with a screen to verify that it should be deleted
**And** if the user chooses Yes from that screen, the computer should be deleted **only if it is has never been assigned to an employee**

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# Testing Instructions for Change Made

```
git fetch --all
git checkout computers
dotnet run
```

1. Click on "Computers" navigation item and confirm a list of computers appears.
2. Click "Create New" link at the top of the list of computers and confirm form contains fields for Computer Manufacturer, Computer Maker and Purchase Date.
3. Enter test data into fields and submit. View should return to list of computers with the newly entered computer added.
4. Click the details link of a computer in the list and confirm details are displayed.
5. Click the delete link of the computer you just created. You should see a confirmation screen and a delete button. Delete the computer. View should return to list of computers and the computer that was deleted shoud not be in the list.
6. Click the delete link of another computer from the original data inserted into our database. The computer should be assigned an employee. The delete confirmation screen should notify you that the computer has been assigned to an employee and there should be no delete button.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] My changes generate no new warnings
- [X] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by jdbbaugh — APPROVED** [review]
🍺

File: BangazonWorkForceManagement/Controllers/ComputersController.cs [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+                    cmd.Parameters.Add(new SqlParameter("@id", id));
+                    cmd.ExecuteNonQuery();
+                    SqlDataReader reader = cmd.ExecuteReader();
+
+                    while (reader.Read())
+
+                    assignedcomputer = reader.IsDBNull(reader.GetOrdinal("ComputerEmployeeCID")) ? (int?)null : (int?)reader.GetInt32(reader.GetOrdinal("ComputerEmployeeCID"));
+
+                    if (assignedcomputer != null) {
+
```
Comment (askingalot) [inline BangazonWorkForceManagement/Controllers/ComputersController.cs:155]: Refactor this `if/else` to make it simpler.

What about something like...?
> `DisplayDelete = assignedcomputer == null;`

File: BangazonWorkForceManagement/Models/ViewModels/ComputerDeleteFailureViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BangazonWorkForceManagement.Models.ViewModels
+{
+    public class ComputerDeleteFailureViewModel
```
Comment (askingalot) [inline BangazonWorkForceManagement/Models/ViewModels/ComputerDeleteFailureViewModel.cs]: Are you using this?

File: BangazonWorkForceManagement/Models/ViewModels/ComputerDeleteViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+        public DateTime PurchaseDate { get; set; }
+
+
+        public string Make { get; set; }
+
+
+        public string Manufacturer { get; set; }
+
+        public bool DisplayDelete { get; set; }
```
Comment (askingalot) [inline BangazonWorkForceManagement/Models/ViewModels/ComputerDeleteViewModel.cs]: It's best practice to use a verb phrase for boolean values, (e.g. `ShouldDisplayDelete`)

File: BangazonWorkForceManagement/Views/Computers/Create.cshtml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+﻿@model BangazonWorkForceManagement.Models.ViewModels.ComputerCreateViewModel
+
+@{
+    ViewData["Title"] = "Create";
+}
+
+<h1>Create</h1>
+
+<h4>ComputerCreateViewModel</h4>
```
Comment (askingalot) [inline BangazonWorkForceManagement/Views/Computers/Create.cshtml]: This isn't the most clear header for the user to see

File: BangazonWorkForceManagement/Views/Computers/Delete.cshtml [hunk 5]
```diff
@@ -0,0 +1,64 @@
+﻿@model BangazonWorkForceManagement.Models.ViewModels.ComputerDeleteViewModel
+
+@{
+    ViewData["Title"] = "Delete";
+}
+
+<h1>Delete</h1>
+
+@if (Model.DisplayDelete == false)
```
Comment (askingalot) [inline BangazonWorkForceManagement/Views/Computers/Delete.cshtml]: `@if (!Model.DisplayDelete)`

File: BangazonWorkForceManagement/Views/Computers/Delete.cshtml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+{
+    <h3>The computer has been assigned to an employee and can not be deleted.</h3>
+}
+
+@if (Model.DisplayDelete == true)
```
Comment (askingalot) [inline BangazonWorkForceManagement/Views/Computers/Delete.cshtml]: `@if (Model.DisplayDelete)`
...or better yet use an `else`

File: BangazonWorkForceManagement/Views/Computers/Details.cshtml [hunk 7]
```diff
@@ -0,0 +1,49 @@
+﻿@model BangazonWorkForceManagement.Models.ViewModels.ComputerDetailViewModel
+
+@{
+    ViewData["Title"] = "Details";
+}
+
+<h1>Details</h1>
+
+<div>
+    <h4>ComputerDetailViewModel</h4>
```
Comment (askingalot) [inline BangazonWorkForceManagement/Views/Computers/Details.cshtml]: Header

**Review by askingalot — CHANGES_REQUESTED** [review]

**Review by asiamcarter — APPROVED** [review]

**Pull request merged**